tests/components/ipp/test_config_flow.py

```diff
@@ -0,0 +1,306 @@
+    aioclient_mock.post(
+        "http://EPSON123456.local:631/ipp/print",
+        content=load_fixture_binary("ipp/get-printer-attributes.bin"),
+        headers={"Content-Type": "application/ipp"},
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": SOURCE_USER},
+    )
+
+    assert result["step_id"] == "user"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_HOST: "EPSON123456.local", CONF_BASE_PATH: "/ipp/print"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "EPSON123456.local"
+
+    assert result["data"]
+    assert result["data"][CONF_HOST] == "EPSON123456.local"
+    assert result["data"][CONF_UUID] == "cfe92100-67c4-11d4-a45f-f8d027761251"
+
+
+async def test_full_zeroconf_flow_implementation(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the full manual user flow from start to finish."""
+    aioclient_mock.post(
+        "http://EPSON123456.local:631/ipp/print",
+        content=load_fixture_binary("ipp/get-printer-attributes.bin"),
+        headers={"Content-Type": "application/ipp"},
+    )
+
+    flow = config_flow.IPPFlowHandler()
+    flow.hass = hass
+    flow.context = {"source": SOURCE_ZEROCONF}
+
+    discovery_info = MOCK_ZEROCONF_IPP_SERVICE_INFO.copy()
+    result = await flow.async_step_zeroconf(discovery_info)
+
+    assert flow.discovery_info
+    assert flow.discovery_info[CONF_HOST] == "EPSON123456.local"
+    assert flow.discovery_info[CONF_NAME] == "EPSON123456"
+
+    assert result["step_id"] == "zeroconf_confirm"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+    result = await flow.async_step_zeroconf_confirm(
+        user_input={CONF_HOST: "EPSON123456.local"}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "EPSON123456"
+
+    assert result["data"]
+    assert result["data"][CONF_HOST] == "EPSON123456.local"
+    assert result["data"][CONF_UUID] == "cfe92100-67c4-11d4-a45f-f8d027761251"
+    assert not result["data"][CONF_SSL]
+
+
+async def test_full_zeroconf_tls_flow_implementation(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the full manual user flow from start to finish."""
+    aioclient_mock.post(
+        "https://EPSON123456.local:631/ipp/print",
+        content=load_fixture_binary("ipp/get-printer-attributes.bin"),
+        headers={"Content-Type": "application/ipp"},
+    )
+
+    flow = config_flow.IPPFlowHandler()
+    flow.hass = hass
+    flow.context = {"source": SOURCE_ZEROCONF}
+
+    discovery_info = MOCK_ZEROCONF_IPPS_SERVICE_INFO.copy()
+    result = await flow.async_step_zeroconf(discovery_info)
+
+    assert result["step_id"] == "zeroconf_confirm"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5274251)